### PR TITLE
Remove custom provider and use javax Provider

### PR DESCRIPTION
**Pull request by nxtstep**
## Description

Use the commonly known `javax.Provider` as a best practise and maybe even lead to an easier adoption 🤷 

## Review checklist

- [x] PR is split into meaningful commits for the ease of reviewing
- [ ] Description contains clear instructions on how to test the feature (where applicable)
- [x] Tests have been written
- [ ] Appropriate labels have been applied

### CODE REVIEW

**Review by hopeman15 — COMMENTED** [review]
One could do this of course... Originally I thought it was quite nice just to have the one dependency with everything pre-packaged, which is why I opted out of using the `javax`s provider. Now, one would need to add the `javax` dependency to make use of custom providers 🤔 If we go this route then let's update the provider section of the readme to reflect this change, and potentially add a migration guide 🙈

**Comment by nxtstep** [comment]
Well, we could also expose it as an `api` dependency, so users would get the (javax.inject) dependency transitively

**Comment by hopeman15** [comment]
> Well, we could also expose it as an `api` dependency, so users would get the (javax.inject) dependency transitively

I'd be okay with that 👍

**Comment by hopeman15** [comment]
What's the status on this one 🤔

File: forge-core/build.gradle.kts [hunk 1]
```diff
@@ -16,14 +16,16 @@ repositories {
 }
 
 dependencies {
-    implementation("org.objenesis:objenesis:3.2")
-    implementation("com.thedeanda:lorem:2.1")
+    implementation(Dependency.objenesis)
+    implementation(Dependency.lorem)
+    // Javax inject
+    implementation(Dependency.JavaX.inject)
```
Comment (nxtstep) [inline forge-core/build.gradle.kts:22]:
```suggestion
    api(Dependency.JavaX.inject)
```

**Review by hopeman15 — APPROVED** [review]
LGTM 👍

**Pull request merged**